xchange-lgo/src/main/java/info/bitrich/xchangestream/lgo/LgoStreamingTradeService.java

```diff
@@ -0,0 +1,287 @@
+        matchedOrder.setOrderStatus(Order.OrderStatus.PARTIALLY_FILLED);
+        matchedOrder.setCumulativeAmount(matchedOrder.getOriginalAmount().subtract(orderEvent.getRemainingQuantity()));
+        BigDecimal fee = matchedOrder.getFee() == null ? orderEvent.getFees() : matchedOrder.getFee().add(orderEvent.getFees());
+        matchedOrder.setFee(fee);
+        return copyOrder(matchedOrder);
+    }
+
+    private Order openOrder(CurrencyPair currencyPair, LgoUserUpdateData orderEvent) {
+        Order pendingOrder = allOrders.get(currencyPair).get(orderEvent.getOrderId());
+        Order.OrderStatus status = pendingOrder.getStatus().equals(Order.OrderStatus.PARTIALLY_FILLED) ? pendingOrder.getStatus() : Order.OrderStatus.NEW;
+        pendingOrder.setOrderStatus(status);
+        return copyOrder(pendingOrder);
+    }
+
+    private Order addPendingOrder(CurrencyPair currencyPair, LgoUserUpdateData orderEvent) {
+        Order order = LgoAdapter.adaptPendingOrder(orderEvent, currencyPair);
+        allOrders.get(currencyPair).put(order.getId(), order);
+        return copyOrder(order);
+    }
+
+    private Order copyOrder(Order order) {
+        Order copy = order instanceof LimitOrder ? LimitOrder.Builder.from(order).build() : MarketOrder.Builder.from(order).build();
+        // because actual released version of xchange-core has buggy Builder.from methods
+        copy.setFee(order.getFee());
+        copy.setCumulativeAmount(order.getCumulativeAmount());
+        // https://github.com/knowm/XChange/pull/3163
+        return copy;
+    }
+
+    @Override
+    public Observable<UserTrade> getUserTrades(CurrencyPair currencyPair, Object... args) {
+        return getBatchOrderEvents(currencyPair)
+                .filter(lgoOrderEvent -> lgoOrderEvent.getType().equals("match"))
+                .map(matchEvent -> LgoAdapter.adaptUserTrade(currencyPair, (LgoMatchOrderEvent) matchEvent));
+    }
+
+    /**
+     * Receive all events for the selected currency pairs. Merges batch order events and ack (AFR)
+     * events.
+     */
+    public Observable<LgoOrderEvent> getAllOrderEvents(Collection<CurrencyPair> currencyPairs) {
+        Observable<LgoOrderEvent> ackObservable = getReceivedOrderEvents();
+        Optional<Observable<LgoOrderEvent>> batchObservable = currencyPairs.stream()
+                .map(this::getBatchOrderEvents)
+                .reduce(Observable::mergeWith);
+        return batchObservable.isPresent() ? ackObservable.mergeWith(batchObservable.get()) : ackObservable;
+    }
+
+    /**
+     * Get ack for your placed orders. "received" events indicate the orderId associated to your
+     * order, if you set a reference on order placement you will have it in the event. "failed" events
+     * indicate that the order could not be read or was invalid and not added to a batch.
+     */
+    public Observable<LgoOrderEvent> getReceivedOrderEvents() {
```

Comment:
Rename to `getRawReceivedOrderEvents` in line with convention

Reply:
https://github.com/bitrich-info/xchange-stream/pull/385/commits/bf18a67a4e7b047677722216dbbf794927610689